packages/tasit-action/src/TransactionSubscription.js

```diff
@@ -11,19 +11,46 @@ export class TransactionSubscription extends Subscription {
   #txConfirmations;
   #timeout;
   #lastConfirmationTime;
-
-  constructor(txPromise, provider) {
+  #contract;
+
+  // Contract passing itself to here, enables
+  // TransactionSubscription calling this.#contract._emitErrorEvent
+  // to emit error to the contract catch-all error listener
+  // As that's is the only function called for now, could be better
+  // pass that function as a callback here?
+  constructor(txPromise, provider, contract) {
     // Provider implements EventEmitter API and it's enough
     //  to handle with transactions events
     super(provider);
 
     const { events } = config;
     const { timeout } = events;
 
+    this.#txPromise = txPromise.then(
+      tx => {
+        return tx;
+      },
+      error => {
+        this._emitErrorEvent(
+          new Error(`Action with error: ${error.message}`),
+          null
```

Comment:
I forget, what's the param that is getting a `null` arg here? `eventName`?

Reply:
Yes, `eventName`

Reply:
Maybe `_emitErrorEvent` shouldn't assume most errors come from other event listeners by default.

What if we make a new function `_emitErrorFromListenerErrorEvent` that takes an error and an `eventName`, and then decorates the error object itself with the eventName key/value and then emits it using `_emitErrorEvent` under the hood?

Reply:
Done

Reply:
Thanks! Resolving.